api/turing/imagebuilder/imagebuilder.go

```diff
@@ -356,3 +358,26 @@ func (ib *imageBuilder) GetImageBuildingJobStatus(
 
 	return JobStatusUnknown, nil
 }
+
+func (ib *imageBuilder) DeleteImageBuildingJob(
+	projectName string,
+	modelName string,
+	versionID models.ID,
+) error {
+	kanikoJobName := ib.nameGenerator.generateBuilderJobName(
```

Comment:
I wonder if the job name should be stored in the DB as part of the job submission process. Otherwise it leaves a chance for some dangling jobs (scenario: job submitted -> new release of Turing is deployed, where `nameGenerator` implementation is changed -> job is terminated)

Reply:
I think its nice to have that, but there are some considerations:

1. What is the expected life time of the job? If the nameGenerator implementation changes, is it okay to let the termination job fail?
2. Are we expecting to change the naming convention?

Reply:
> Are we expecting to change the naming convention?

Not "expecting", but this can happen for some reason (intentionally or accidentally) and whoever changes the nameGenerator might not even be aware of the implications. 

> is it okay to let the termination job fail?

What is `termination job`?

Reply:
Well I mean the Termination request from the API.

Reply:
okay